provider_config_test.go

```diff
@@ -0,0 +1,122 @@
+package jwtauth
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type testProviderConfig struct {
+	initialized string
+	throwError  bool
+}
+
+func (t *testProviderConfig) Initialize(jc *jwtConfig) error {
+	if t.throwError {
+		return fmt.Errorf("i'm throwing an error")
+	}
+	t.initialized = jc.ProviderConfig["initialized_value"].(string)
+	return nil
+}
+
+func (t *testProviderConfig) SensitiveKeys() []string {
+	return []string{"these", "are", "secret"}
+}
+
+func TestNewProviderConfig(t *testing.T) {
+
+	t.Run("normal case", func(t *testing.T) {
+		jc := &jwtConfig{
+			ProviderConfig: map[string]interface{}{
+				"provider":          "test",
+				"initialized_value": "yes",
+			},
+		}
+		pMap := map[string]CustomProvider{
+			"test": &testProviderConfig{},
+			// "gsuite": &GsuiteProvider{},
+			// "azure":  &AzureProvider{},
```

Comment:
Are these placeholders, examples, or foreshadowing 😆 ?

Reply:
Foreshadowing examples. Watch this space!

Reply:
Removed in fe083e8